backend/node/generate_nodes.py

```diff
@@ -0,0 +1,65 @@
+import json
+import os
+import secrets
+
+
+def generate_secret():
+    return secrets.token_hex(64)
+
+
+def generate(orgs):
+    files = {}
+
+    # TODO merge two loops
+    # init file content
+    for org in orgs:
+        data = {
+            'incoming_nodes': [],
+            'outgoing_nodes': [],
+        }
+        files[org] = data
+
+    for org in orgs:
+        # create intern node (request from worker A to backend A)
+        secret = generate_secret()
+        files[org]['outgoing_nodes'].append({
+            'node_id': org,
+            'secret': secret
+        })
+        files[org]['incoming_nodes'].append({
+            'node_id': org,
+            'secret': secret
+        })
+
+        for other_org in filter(lambda x: x != org, orgs):
+            # outgoing from server B to server A share same secret as incoming from server B in server A
+            secret = generate_secret()
+            files[other_org]['outgoing_nodes'].append({  # in server B
+                'node_id': org,  # to server A
+                'secret': secret
+            })
+
+            files[org]['incoming_nodes'].append({  # in server A
+                'node_id': other_org,  # from server B
+                'secret': secret
+            })
+
+    return files
+
+
+def generate_for_orgs(orgs):
+    files = generate(orgs)
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    nodes_path = os.path.join(dir_path, 'nodes')
+    os.makedirs(nodes_path, exist_ok=True)
+    for k, v in files.items():
+        filepath = os.path.join(nodes_path, f'{k}.json')
+        with open(filepath, 'w') as f:
+            f.write(json.dumps(v, indent=4))
+
+
+if __name__ == '__main__':
+    orgs = ['owkinMSP', 'chu-nantesMSP', 'clbMSP']  # TODO should be discovered by discovery service
+
+    generate_for_orgs(orgs)
```

Comment:
I re-read the two "generate" scripts (this one and `scripts/generate_nodes_fixtures.py`) and it really looks like they do the exact same thing. It also looks like the `init_node.py` command is a simpler (because only targeting two models and one use case) version of django's loaddata.

In a previous conversation about these, you said:

> The extra django command is also used for admin modification in the future. When you want to add a new organization, you then will be able to add Incoming/Outgoing Node easily. With fixtures, this is not possible.

I agree that only loading committed fixtures the way we do it now isn't dynamic. But this is because of the "committed" part, not because of the "fixtures". Fixtures can be generated and loaded at any time, with any new content, in the same way you propose here to use the generate_nodes.py / init_nodes.py

Could elaborate on the added benefits of these new scripts? Because if there's no absolute need for them I'd rather we keep the already existing ones.

Reply:
I can understand the feeling, but these are extra changes that are not strictly necessary. 

Since this is already a pretty hefty PR, coupled with other large ones, I really think we should stick to having as few changes as the overall feature requires.

Reply:
I've just thought to another thing about fixtures.
Before starting this PR the `generate_nodes_fixtures.py` script was not present in the code base, only fixtures files were present.
I don't think commiting fixtures files with secret inside is a good practice.

Currently the main code commited these files:
https://github.com/SubstraFoundation/substra-backend/tree/dev/backend/node/fixtures
And refers to the new `generate_nodes_fixtures.py` script: https://github.com/SubstraFoundation/substra-backend/blob/dev/scripts/generate_nodes_fixtures.py

I'm not feeling comfortable by commiting secret even generated from fixtures.
Using dynamic secret not commited feel more secure and clean.

Reply:
That's a good point, thanks.